resources-domain/resources-search-index/tests/integration/resources-search-index.test.ts

```diff
@@ -16,22 +16,27 @@
 
 import { SQS } from 'aws-sdk';
 
-import { Client, IndexTypes, getClient, SearchResponse } from '@tech-matters/elasticsearch-client';
+import { IndexClient, getClient } from '@tech-matters/elasticsearch-client';
+
+import {
```

Comment:
@stephenhand This test was already failing and isn't actually fixed yet. I gave up until we can have a bit more discussion around the split of the ES client into an indexClient and a searchClient and the general layout of the dependency injection search configuration system.

I think the split added a decent amount of complexity to the es client API with minimal short term benefit. I also believe that some of the dependency injection configuration system complexity would be better abstracted away into the client for parts that will likely be shared between indices in future iterations.

I didn't want this PR to grow out of control, so I hit a stopping point of "it transpiles and all of the tests that were passing before still pass". Assuming that testing in development works, I'll add tests and potentially a bit of refactoring based on our conversations around the index split and where index configuration lines should be driven in a follow up PR, if that works for you.

Reply:
Ok, I'm fine to move more functionality into the core client, I went back and forth but in the end went for a cautious approach about deciding what functionality was likely to be common, thinking we could always move functionality from the injected dependencies to the core client in the future - which is exactly what you're suggesting now, so ok with me

As for the clients, it depends how they look after some of the functionality has moved out of the dependencies. I don't like the idea that you have to provide a bunch of configuration irrelevant to what you're trying to do - the shared configuration between indexing and searching seemed small and, except for testing, it seemed like there were rarely use cases where you are doing both in the same operation. 

But if moving some stuff back to the core seriously reduces the amount of indexing specific and searching specific configuration then recombining them might make more sense to be. 

We can dig into specifics further on a call if you like